pyteal/ast/subroutine.py

```diff
@@ -191,14 +193,13 @@ def _arg_types_and_by_refs(
                         f"Function keyword parameter {name} has type {expected_arg_type}"
                     )
                 abi_output_kwarg[name] = expected_arg_type
-                continue
             else:
                 expected_arg_types.append(expected_arg_type)
 
-            if expected_arg_type is ScratchVar:
-                by_ref_args.add(name)
-            if isinstance(expected_arg_type, abi.TypeSpec):
-                abi_args[name] = expected_arg_type
+                if expected_arg_type is ScratchVar:
+                    by_ref_args.add(name)
+                if isinstance(expected_arg_type, abi.TypeSpec):
+                    abi_args[name] = expected_arg_type
```

Comment:
we can still keep the continue, and remove the `else` to lessen the indentation, like:
```python
# some code before ...
                abi_output_kwarg[name] = expected_arg_type
                continue

            expected_arg_types.append(expected_arg_type)

            if expected_arg_type is ScratchVar:
                by_ref_args.add(name)
            if isinstance(expected_arg_type, abi.TypeSpec):
                abi_args[name] = expected_arg_type
# some code after ...
```

Reply:
@ahangsu Gauging your preference:  Are you saying you _prefer_ the `continue` statement?

* If yes, then I'll change to your preference because I'm presenting the PR optionally.
* Context from my side: 
  * I specifically removed it because I think it complicates comprehension (as noted in PR description).  
  * The indentation did _not_ strike me as a concern.  I can imagine a broader change that removes indentation, but I'm trying to present minimal changes.  Particularly for a refactoring.

Reply:
I am slightly preferring `continue` statement, but they should be having the same, so I have no too much comments on that, and feel free to ignore my previous thoughts.

Reply:
@ahangsu I appreciate the flexibility though I'm only looking to merge if you prefer the changes.  Via https://github.com/algorand/pyteal/pull/308/commits/134e10631eb00d4f40b3b6ab780b1adc356b41cc, I switched to the suggestion you raised earlier.

If it looks good to you, let's resolve the thread.

Reply:
lgtm, im resolving!